processor/tailsamplingprocessor/and_helper.go

```diff
@@ -15,25 +15,25 @@
 package tailsamplingprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor"
 
 import (
-	"go.uber.org/zap"
+	"go.opentelemetry.io/collector/component"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor/internal/sampling"
 )
 
-func getNewAndPolicy(logger *zap.Logger, config *AndCfg) (sampling.PolicyEvaluator, error) {
+func getNewAndPolicy(settings component.TelemetrySettings, config *AndCfg) (sampling.PolicyEvaluator, error) {
```

Comment:
This could have been split from this PR, making it easier to review/merge.

Reply:
True and it still could be broken out if you want

Reply:
+1 OK to split. I will keep in mind that we should do exactly one thing in a single PR. 

Since we got many reviews in this PR already, reserving this change might save some time and efforts of our reviewer?

Reply:
Ya we'll leave it.